src/ui/castleinternaljoystickswindows.pas

```diff
@@ -283,8 +293,20 @@ procedure TWindowsJoysticksBackend.Poll(const List: TJoystickList;
         Joystick.State.BtnUp[ j ] := btn = 0;
         if Assigned(EventContainer.OnButtonDown) and (btn <> 0) then EventContainer.OnButtonDown(Joystick, j);
       end;
+    end else
+    if JoyError = JOYERR_UNPLUGGED then
+    begin
+      WritelnLog('Joystick was unplugged. Trying to run Joysticks.Initialize.');
+      JoystickHadBeenDisconnected := true;
+      { Note that only last unplugged joystick will be sent to the Unplugged event
+        if multiple joysticks have been unplugged simultaneously
+        which may happen e.g. in case an USB-Hub with several joysticks had been disconnected }
+      JoystickDisconnected := Joystick;
```

Comment:
Same comment as above for Linux: let's not remember/pass `JoystickDisconnected`.